### PR TITLE
[api] Fix Object<T> in view functions

### PR DESCRIPTION
**Pull request by gregnazario**
### Description

Fixes View functions to work with `Object<T>` as well as  demo for it.

### <samp>🤖 Generated by Copilot at c22199c</samp>

This pull request adds support for generic token wrappers (`Object` types) to the `aptos-core` API and the `aptos-move` language. It implements type conversion, type representation, and example usage of `Object` types in the `api/types` and `aptos-move/move-examples` crates. It also improves some existing code quality and functionality in the `hero` module.

### Test Plan
Using a normal output works:
```
$ aptos move view --profile local --function-id local::hero::view_hero --args address:local 'string:Hero Quest!' 'string:Bob'
{
  "Result": [
    "male",
    "human"
  ]
}
```

Using a hardcoded T works:

```
$ aptos move view --profile local --function-id
 local::hero::view_hero_by_object --args address:0xd4a36e876ba601f4690bade133947bb328246ed63e1a83d96829a66f909a4dcc       
{
  "Result": [
    "male",
    "human"
  ]
}
```

Using arbitrary `T` only works if the `T` is the correct type.
```
$ aptos move view --profile local --function-id local::hero::view_object --args address:0xd4a36e876ba601f4690bade133947bb328246ed63e1a83d96829a66f909a4dcc --type-args 0xb98022721f97da65950435386a5209be2fc54bed5f12fc8ea0a1c3d0c76f562f::hero::Hera
{
  "Error": "API error: API error Error(InvalidInput): VMError with status TYPE_RESOLUTION_FAILURE at location UNDEFINED and message Cannot find ModuleId { address: b98022721f97da65950435386a5209be2fc54bed5f12fc8ea0a1c3d0c76f562f, name: Identifier(\"hero\") }::IdentStr(\"Hera\") in cache"
}

$ aptos move view --profile local --function-id local::hero::view_object --args address:0xd4a36e876ba601f4690bade133947bb328246ed63e1a83d96829a66f909a4dcc --type-args 0xb98022721f97da65950435386a5209be2fc54bed5f12fc8ea0a1c3d0c76f562f::hero::Hero
{
  "Result": [
    "male",
    "human"
  ]
}

$ aptos move view --profile local --function-id local::hero::view_object --args address:0xd4a36e876ba601f4690bade133947bb328246ed63e1a83d96829a66f909a4dcc --type-args u8                                                                           
{
  "Error": "API error: API error Error(InvalidInput): VMError with status CONSTRAINT_NOT_SATISFIED at location Module ModuleId { address: b98022721f97da65950435386a5209be2fc54bed5f12fc8ea0a1c3d0c76f562f, name: Identifier(\"hero\") }"
}

```

### CODE REVIEW

**Review by gerben-stavenga — APPROVED** [review]

File: api/types/src/convert.rs [hunk 1]
```diff
@@ -47,9 +48,13 @@ use std::{
     convert::{TryFrom, TryInto},
     iter::IntoIterator,
     rc::Rc,
+    str::FromStr,
     sync::Arc,
 };
 
+static OBJECT_MODULE: Lazy<Identifier> = Lazy::new(|| Identifier::from_str("object").unwrap());
```
Comment (gerben-stavenga) [inline api/types/src/convert.rs:55]: Can't you remove the lazy and make it IdentStr

Comment (davidiw) [reply api/types/src/convert.rs:55]: isn't the normal approach to add this type of things to the types crate?
where every one of these is done this way :P

File: api/types/src/convert.rs [hunk 2]
```diff
@@ -754,7 +759,22 @@ impl<'a, R: MoveResolverExt + ?Sized> MoveConverter<'a, R> {
         type_tag: &TypeTag,
         val: Value,
     ) -> Result<move_core_types::value::MoveValue> {
-        let layout = self.inner.get_type_layout_with_types(type_tag)?;
+        let layout = match type_tag {
+            TypeTag::Struct(boxed_struct) => {
+                // This has to be checked at this level, otherwise it goes into the Move language
```
Comment (davidiw) [inline api/types/src/convert.rs:764]: might rephrase that the current framework cannot parse generic layouts so we perform this here

File: aptos-move/move-examples/token_objects/hero/sources/hero.move [hunk 3]
```diff
@@ -210,18 +211,57 @@ module token_objects::hero {
...
 
+    // View functions
+    #[view]
+    fun view_hero(creator: address, collection: String, name: String): (String, String) acquires Hero {
+        let (_, hero) = get_hero(
+           &creator,
+            &collection,
+            &name,
+        );
+        (hero.gender, hero.race)
```
Comment (davidiw) [inline aptos-move/move-examples/token_objects/hero/sources/hero.move:232]: we can't view the hero struct?

**Review by davidiw — APPROVED** [review]

File: aptos-move/move-examples/token_objects/hero/sources/hero.move (hunk 3 above)
Comment (gregnazario) [reply aptos-move/move-examples/token_objects/hero/sources/hero.move:232]: I guess we could, I didn't try that

File: api/types/src/convert.rs (hunk 1 above)
Comment (gregnazario) [reply api/types/src/convert.rs:55]: Looks like someone made `ident_str!` allowing them as compile time constants

**Pull request merged**